app/mailers/discovered_mailer.rb

```diff
@@ -1,7 +1,18 @@
 class DiscoveredMailer < ::ApplicationMailer
   helper :discovered_hosts
   def discovered_summary(options = {})
-    raise ::Foreman::Exception.new(N_("Must specify a user with email enabled")) unless (user = User.find(options[:user])) && user.mail_enabled?
+    user = if options[:user].kind_of? User
+             options[:user]
+           elsif options[:user].kind_of? Integer
+             User.find(options[:user])
+           else
+             raise ::Foreman::Exception.new(_('Invalid provided user type of %s') % options[:user].class.to_s)
```

Comment:
Incorrect use of exception, please read: https://projects.theforeman.org/projects/foreman/wiki/Translating#Exceptions

Reply:
Oh, I was wondering why some uses the `N_` method on some places but not always, thanks!